File: 진우/11weeks/1620. 나는야 포켓몬 마스터 이다솜/나는야 포켓몬 마스터 이다솜.cc
#include <iostream>
#include <string>
#include <vector>
#include <map>
#include <algorithm>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    
    int N, M;
    cin >> N >> M;

    string name;

    // 문자열 이름 → 번호 저장
    map<string, int> pokedex;

    // 번호 → 이름 저장 (번호로 이름을 찾기 위해 사용)
    vector<string> num(N + 1); // 1번부터 사용하기 위해 N+1 크기

    // 포켓몬 이름 입력 (1번부터 N번까지)
    for (int i = 1; i <= N; i++) {
        cin >> name;
        pokedex[name] = i;  // 이름 → 번호 매핑
        num[i] = name;      // 번호 → 이름 매핑
    }

    // M개의 질문 처리
    for (int i = 0; i < M; i++) {
        cin >> name;

        // 첫 글자가 숫자인지 확인 → 숫자로 시작하면 번호 입력
        if ('0' <= name[0] && name[0] <= '9') {
            int idx = stoi(name);  // 문자열을 정수로 변환
            cout << num[idx] << "\n";  // 번호로 이름 출력
        }
        else {
            // 문자열 입력 → 이름으로 번호 찾기
            cout << pokedex[name] << "\n";
        }
    }

    return 0;
}

File: 진우/11weeks/1620. 나는야 포켓몬 마스터 이다솜/나는야 포켓몬 마스터 이다솜.py
import sys
input = sys.stdin.readline

# N: 포켓몬 수, M: 문제에서 물어볼 횟수
N, M = map(int, input().split())

# 이름 → 번호 매핑용 딕셔너리
pokedex = {}

# 번호 → 이름 매핑용 리스트 (1번부터 사용하므로 N+1 크기)
num = [""] * (N + 1)

# 포켓몬 도감 데이터 입력
for i in range(1, N + 1):
    name = input().strip()  # 포켓몬 이름 입력
    pokedex[name] = i       # 이름으로 번호 찾기
    num[i] = name           # 번호로 이름 찾기

# 문제 질의 처리
for _ in range(M):
    name = input().strip()  # 입력(이름 또는 번호)

    # 첫 글자가 숫자면 → 번호로 간주
    if '0' <= name[0] <= '9':
        idx = int(name)     # 문자열 숫자를 정수로 변환
        print(num[idx])     # 번호에 해당하는 포켓몬 이름 출력
    else:
        # 숫자가 아니면 → 이름으로 간주
        print(pokedex[name])   # 이름에 해당하는 번호 출력
